woodwork/tests/type_system/conftest.py

```diff
@@ -150,6 +139,9 @@ def bad_pandas_emails():
         pd.Series(["fl@alteryx.com", "b☃d@email.com", "good@email.com", np.nan]),
         pd.Series(["fl@alteryx.com", "@email.com", "good@email.com", "foo@bar.com"]),
         pd.Series(["fl@alteryx.com", "bad@email", "good@email.com", np.nan]),
+        pd.Series([1, 2, 3, 4]).astype("int"),
+        pd.Series([{"key": "value"}]).astype("O"),
+        pd.Series([(1, 2), (3, 4)]).astype("O"),
```

Comment:
Minor point, but can we update these new series so that these are similar to the series we use above, where we have some valid email addresses, and only a single bad value? Our inference should be setup to fail if only a single value fails to match the regex, and having only a single bad value might make that a little more obvious.

Reply:
I can add those cases as well, but the original issue was a series of tuples-only and that was getting inferred as an email address, so it seems like we would want to check that explicitly maybe?

Reply:
When the issue was discovered, it just happened to be with a series of all tuples, but there wasn't really anything special about that input series other than the fact that it contained tuples. We can check both situations, but generally I don't think we need to go too far with creating all kinds of different combinations of inputs here.

Reply:
After discussion with @dvreed77 mixing tuples and strings creates issues for the Koalas tests, so will leave this as it is. The changes in this PR should still prevent a column with valid email addresses and a single tuple from being inferred as an email address.